test/cpp/api/dataloader.cpp

```diff
@@ -1880,4 +1914,205 @@ TEST(DataLoaderTest, ChunkDatasetDoesNotHang) {
   // to fill the batch buffer but it is not draining. Still we need to exit
   // cleanly.
   auto iterator = data_loader->begin();
-}
\ No newline at end of file
+}
+
+TEST(DataLoaderTest, ChunkDatasetSaveOption) {
+  const size_t chunk_count_ = 6;
+  const size_t chunk_size = 10;
+
+  struct DummyTestChunkDataReader : datasets::ChunkDataReader<int> {
```

Comment:
Don't we have a Dummy reader already in this file? please check.

Reply:
The Dummy reader defines chunk size to be different to each other (3 chunks with size equal to 5, 10, 20). Here I need to set the cache size such that at most one chunk is stored in the cache. With Dummy readers config, it is hard to achieve this. So I made another internal reader just for this test.

Reply:
Thank you for the clarification.